### PR TITLE
Api 39033 extract claims validation into service

### PR DESCRIPTION
**Pull request by rockwellwindsor-va**
## Summary
* Moves the validate method (and one dependent method) into a service
* Adds tests for methods
* Additional Fixes
	* Fixes a typo in related test text
	* Fixes a variable in related test to correctly be a string value

## Related issue(s)
[API-39033](https://jira.devops.va.gov/browse/API-39033)

## Testing done
- [x] *New code is covered by unit tests*
-  For what it is worth:
Screenshot 2024-08-20 at 10 15 46 AM

#### Testing notes
* This is used in the claims show process.
   * To view accessing claims as a dependent `http://localhost:3000/services/claims/v2/veterans/1012667057V811306/claims/600123962`
   * Otherwise, just access claims for the target veteran to test.  
        * Should have access `http://localhost:3000/services/claims/v2/veterans/1012667057V811306/claims/600537706 `
        * Should not have access (Janet Moore ICN for MArgie Curtis claim): `http://localhost:3000/services/claims/v2/veterans/1012861229V078999/claims/600123962`
        * Returns: Screenshot 2024-08-20 at 10 21 23 AM
* Outside of the suggested examples you can put in any ICN into the GET claims endpoint to get claim IDs.  Then try to access the claims show using those IDs and a different ICN.

## What areas of the site does it impact?
        created:   modules/claims_api/lib/claims_api/v2/claims/claim_validator.rb
	modified:   modules/claims_api/spec/factories/auto_established_claims.rb
	created:   modules/claims_api/spec/lib/claims_api/v2/claim_validator_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/claims_api/spec/factories/auto_established_claims.rb [hunk 1]
```diff
@@ -241,7 +241,7 @@
     phase_chngd_dt { Faker::Time.backward(days: 5, period: :morning) }
     phase_type { 'Pending Decision Approval' }
     phase_type_change_ind { '76' }
-    ptcpnt_vet_id { Faker::Number.number(digits: 9) }
+    ptcpnt_vet_id { Faker::Number.number(digits: 9).to_s }
```
Comment (rockwellwindsor-va) [inline modules/claims_api/spec/factories/auto_established_claims.rb:244]: This is a fix for a related test.  This value on a claim is a string so just want to make sure all the data is correctly set (I cannot speak to the other examples in here that do not do this and what that may mean for those tests).

File: modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb [hunk 2]
```diff
@@ -548,7 +548,7 @@
       end
     end
 
-    context 'show with validate_id_with_icn when there is a claimant ID in place of the verteran ID' do
+    context 'show with validate_id_with_icn when there is a claimant ID in place of the veteran ID' do
```
Comment (rockwellwindsor-va) [inline modules/claims_api/spec/requests/v2/veterans/claims_request_spec.rb:551]: fixing typo for related test to this work

**Comment by stiehlrod** [comment]
@rockwellwindsor-va In your testing examples, for number 2, did you mean to paste in a different ICN & claim ID?
Screenshot 2024-08-20 at 12 36 18 PM

**Comment by rockwellwindsor-va** [comment]
> @rockwellwindsor-va In your testing examples, for number 2, did you mean to paste in a different ICN & claim ID? Screenshot 2024-08-20 at 12 36 18 PM

I did, that is showing it catch the error if they shouldn't access the claim

**Comment by stiehlrod** [comment]
> * Should have access `http://localhost:3000/services/claims/v2/veterans/1012667057V811306/claims/600543205`

@rockwellwindsor-va Maybe I am confused, I thought the one that should have access would not return a resource not found

**Review by mchristiansonVA — APPROVED** [review]
Tested locally via Postman and confirmed /show handling for claims associated with ICN as well as not associated with ICN. Looks good to me!

**Pull request closed without merging**